File: backend/app/api/patients.py
from fastapi import APIRouter, Query
from app.models.patient import PatientCreate, PatientUpdate
from app.services import patient_service

router = APIRouter(prefix="/patients", tags=["Patients"])


@router.post("/", status_code=201)
async def create_patient(body: PatientCreate):
    return await patient_service.create_patient(body)


@router.get("/")
async def list_patients(
    skip: int = Query(0, ge=0),
    limit: int = Query(50, ge=1, le=200),
):
    return await patient_service.list_patients(skip=skip, limit=limit)


@router.get("/{patient_id}")
async def get_patient(patient_id: str):
    return await patient_service.get_patient(patient_id)


@router.put("/{patient_id}")
async def update_patient(patient_id: str, body: PatientUpdate):
    return await patient_service.update_patient(patient_id, body)


@router.delete("/{patient_id}")
async def delete_patient(patient_id: str):
    return await patient_service.delete_patient(patient_id)

File: frontend/src/api/patients.js
import client from './client'

export const getPatients = (skip = 0, limit = 100) =>
  client.get(`/patients/?skip=${skip}&limit=${limit}`).then(r => r.data)

export const getPatient = (id) =>
  client.get(`/patients/${id}`).then(r => r.data)

export const createPatient = (data) =>
  client.post('/patients/', data).then(r => r.data)

export const updatePatient = (id, data) =>
  client.put(`/patients/${id}`, data).then(r => r.data)

export const deletePatient = (id) =>
  client.delete(`/patients/${id}`).then(r => r.data)

export const getVisits = (patientId) =>
  client.get(`/visits/patient/${patientId}`).then(r => r.data)

export const createVisit = (data) =>
  client.post('/visits/', data).then(r => r.data)

export const getDocuments = (patientId) =>
  client.get(`/documents/patient/${patientId}`).then(r => r.data)

export const uploadDocument = (formData) =>
  client.post('/documents/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  }).then(r => r.data)

export const getPatientSummary = (patientId) =>
  client.post('/ai/patient-summary', { patient_id: patientId }).then(r => r.data)

export const ragQuery = (query, patientId = null) =>
  client.post('/ai/query', { query, patient_id: patientId }).then(r => r.data)

export const predictNoShow = (data) =>
  client.post('/ai/predict-noshow', data).then(r => r.data)
